zulipterminal/core.py

```diff
@@ -617,11 +617,18 @@ def narrow_to_all_mentions(self) -> None:
         self._narrow_to(anchor=None, mentioned=True)
 
     def deregister_client(self) -> None:
-        queue_id = self.model.queue_id
-        self.client.deregister(queue_id, 1.0)
+        raise urwid.ExitMainLoop
```

Comment:
This does seem cleaner than the sys.exit we currently use.

However, without that I've experienced strange states where the client doesn't exit properly.

Reply:
Thank you for your comment and for reviewing the code. I appreciate your feedback.
I've reverted to the existing code, the behavior seems apt now.